### PR TITLE
removing one by one hmis profile (DEV-842)

### PR DESCRIPTION
**Pull request by Davit-BetterAngels**
DEV-842

## Summary by Sourcery

Enhance the HMIS profiles management by allowing users to remove or reset individual profiles within the form.

Enhancements:
- Add functionality to remove individual HMIS profiles from the list.
- Introduce a reset option for individual HMIS profiles to clear specific fields.

### CODE REVIEW

**Review by johnr54321 — APPROVED** [review]
Tested on andriod and iphone.  Behaves as expected. Approved.

File: libs/expo/betterangels/src/lib/screens/AddEditClient/PersonalInfo/HmisProfiles.tsx [hunk 1]
```diff
@@ -35,14 +37,19 @@ export default function HmisProfiles() {
     setValue('hmisProfiles', []);
   };
 
+  const onItemReset = (index: number) => {
+    resetField(`hmisProfiles.${index}.agency`);
+    resetField(`hmisProfiles.${index}.hmisId`);
```
Comment (tglaz) [inline libs/expo/betterangels/src/lib/screens/AddEditClient/PersonalInfo/HmisProfiles.tsx:42]: i'm seeing strange behavior when clicking a `reset` button. 


1. change ID#
2. click Reset (nothing seems to happen)
3. change ID (lahsa -> some other)
4. ID # now resets (to original value)

Not sure if it's due referencing by `index` here vs using `keyId`.
Or maybe related to not passing `defaultValue` as option, or not setting it?. Not sure.

File: libs/expo/betterangels/src/lib/screens/AddEditClient/PersonalInfo/HmisProfiles.tsx [hunk 2]
```diff
@@ -22,11 +22,13 @@ export default function HmisProfiles() {
...
     control,
     formState: { errors },
+    resetField,
   } = useFormContext<UpdateClientProfileInput | CreateClientProfileInput>();
 
-  const { fields, append } = useFieldArray({
+  const { fields, append, remove } = useFieldArray({
     control,
     name: 'hmisProfiles',
+    keyName: 'keyId',
```
Comment (tglaz) [inline libs/expo/betterangels/src/lib/screens/AddEditClient/PersonalInfo/HmisProfiles.tsx:31]: not sure you saw this, but from their docs about `keyName`:
`this prop is no longer required and will be removed in the next major version.`

so ok for now, but FYI

**Review by tglaz — CHANGES_REQUESTED** [review]

**Comment by johnr54321** [comment]
Ok to have just remove.  Reset is in a separate ticket (Dev-884)

**Review by tglaz — APPROVED** [review]

**Pull request merged**